components/ActivityDetails/index.js

```diff
@@ -1,42 +1,120 @@
 import StyledImageComponent from "@/components/StyledImageComponent";
-import { StyledLink } from "@/components/StyledLinks";
+import { TransparentBackLink } from "@/components/StyledLinks";
 import { Icon } from "@/components/Icon";
 import styled from "styled-components";
 import { StyledButton } from "../StyledButton";
+import { useRouter } from "next/router";
 
 export default function ActivityDetails({ activity, onDeleteActivity }) {
+  const router = useRouter();
+
+  const handleEditClick = () => {
+    router.push(`/${activity.id}/edit`);
+  };
+
   return (
-    <article>
-      <StyledImageComponent src={activity.image} alt={activity.title} />
-      <StyledDiv>
-        <h2>{activity.title}</h2>
-        <div>
-          <StyledLink href={`/${activity.id}/edit`}>
-            <Icon name="edit" />
-          </StyledLink>
-          <StyledButton
-            type="button"
-            $variant="delete"
-            onClick={() => onDeleteActivity(activity.id)}
-          >
-            <Icon name="delete" />
-          </StyledButton>
-        </div>
-      </StyledDiv>
-      <div>
-        <p>
-          {activity.area}, {activity.country}
-        </p>
-        <p>{activity.category.join(", ")}</p>
-      </div>
-      <p>{activity.description}</p>
-    </article>
+    <StyledArticle>
+      <ImageContainer>
+        <TransparentBackLink href="/" title="Back to all Activities">
+          <Icon name="chevronLeft" color="black" />
+        </TransparentBackLink>
+        <StyledImageComponent src={activity.image} alt={activity.title} />
+      </ImageContainer>
+      <Content>
+        <StyledDiv>
+          <Title>{activity.title}</Title>
+          <ActionIcons>
+            <SmallButton
+              title="Edit activity"
+              type="button"
+              $variant="edit"
+              onClick={handleEditClick}
+            >
+              <Icon name="edit" />
+            </SmallButton>
```

Comment:
Why did you went back to a button? This button only makes a router.push() and nothing else. In this case it's best to use a Link. 
Use router.push() when you want to execute some specific logic and push to another site afterwards.